node/core/backing/src/lib.rs

```diff
@@ -511,48 +802,20 @@ async fn validate_and_make_available(
 		},
 	};
 
-	tx_command.send(make_command(res)).await.map_err(Into::into)
+	tx_command.send((relay_parent, make_command(res))).await.map_err(Into::into)
 }
 
 struct ValidatorIndexOutOfBounds;
 
-impl CandidateBackingJob {
-	/// Run asynchronously.
-	async fn run_loop(
-		mut self,
-		mut sender: JobSender<impl SubsystemSender>,
-		mut rx_to: mpsc::Receiver<CandidateBackingMessage>,
-		span: PerLeafSpan,
-	) -> Result<(), Error> {
-		loop {
-			futures::select! {
-				validated_command = self.background_validation.next() => {
-					let _span = span.child("process-validation-result");
-					if let Some(c) = validated_command {
-						self.handle_validated_candidate_command(&span, &mut sender, c).await?;
-					} else {
-						panic!("`self` hasn't dropped and `self` holds a reference to this sender; qed");
-					}
-				}
-				to_job = rx_to.next() => match to_job {
-					None => break,
-					Some(msg) => {
-						// we intentionally want spans created in `process_msg` to descend from the
-						// `span ` which is longer-lived than this ephemeral timing span.
-						let _timing_span = span.child("process-message");
-						self.process_msg(&span, &mut sender, msg).await?;
-					}
-				}
-			}
-		}
-
-		Ok(())
-	}
-
+impl<Context> CandidateBackingJob<Context>
+where
+	Context: SubsystemContext,
+	Context: overseer::SubsystemContext,
+{
 	async fn handle_validated_candidate_command(
 		&mut self,
 		root_span: &jaeger::Span,
-		sender: &mut JobSender<impl SubsystemSender>,
+		ctx: &mut Context,
```

Comment:
I don't think this is necessary, a `sender: impl SubsystemSender` should suffice - this goes a long way

Reply:
The `Context` actually is necessary because of `spawn`. It's not necessary everywhere, but almost everywhere.

Reply:
I missed that `spawn`, thanks for clarifying. The original jobs subsystem impl worked around that by adding another layer of enums, not sure that's any better.